libvcx/src/libindy/proofs/proof_request.rs

```diff
@@ -39,26 +39,31 @@ impl ProofRequestData {
     }
 
     pub fn set_requested_attributes(mut self, requested_attrs: String) -> VcxResult<ProofRequestData> {
-        let requested_attributes: Vec<AttrInfo> = ::serde_json::from_str(&requested_attrs)
-            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Invalid Requested Attributes: {:?}, err: {:?}", requested_attrs, err)))?;
-
-        for attribute in requested_attributes.iter() {
-            if attribute.name.is_some() && attribute.names.is_some() {
-                return Err(VcxError::from_msg(VcxErrorKind::InvalidProofRequest,
-                                              format!("Requested attribute can contain either 'name' or 'names'. Not both.")));
-            };
+        match serde_json::from_str::<HashMap<String, AttrInfo>>(&requested_attrs) {
+            Ok(attrs) => self.requested_attributes = attrs,
+            Err(err) => {
+                warn!("Error when setting attributes (this might not be a problem): {:?}", err);
```

Comment:
I think here shouldn't be any warn/error message at all? Since Err branch just implies possibly the old input format, and we are throwing error on line 47 if even that deserialization fails

Reply:
Trace level would be probably more appropriate, but decided to remove it completely in the end.